controller/node_controller.go

```diff
@@ -323,9 +325,21 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
 		}
 		// get disk available size
 		diskInfo, err := nc.getDiskInfoHandler(disk.Path)
-		if err != nil {
-			logrus.Errorf("Get disk information on node %v error: %v", node.Name, err)
+		// if the file system has been changed or the disk has been umount
+		if err != nil || diskInfo == nil || diskInfo.Fsid != diskID {
```

Comment:
diskInfo won't be nil if err is nil.

Also, print different logs if err is not nil vs fsid changed. One is cannot detect the disk on the path, the other is filesystem has changed.